File: library/src/main/java/com/raizlabs/android/dbflow/structure/BaseQueryModel.java
package com.raizlabs.android.dbflow.structure;

/**
 * Description:
 */
public class BaseQueryModel extends BaseFinalModel {

    @Override
    public boolean exists() {
        throw new InvalidSqlViewOperationException("View " + getClass().getName() + " does not exist");
    }
}

File: dbflow-common/src/main/kotlin/com/raizlabs/dbflow5/structure/BaseQueryModel.kt
package com.raizlabs.dbflow5.structure

import com.raizlabs.dbflow5.annotation.QueryModel
import com.raizlabs.dbflow5.database.DatabaseWrapper

/**
 * Description: Provides a base class for objects that represent [QueryModel].
 */
class BaseQueryModel : NoModificationModel() {

    override fun exists(wrapper: DatabaseWrapper): Boolean {
        throw InvalidSqlViewOperationException("Query ${wrapper::class} does not exist as a table." +
            "It's a convenient representation of a complex SQLite cursor.")
    }
}
